certbot/compat/os.py

```diff
@@ -51,6 +51,28 @@ def chmod(*unused_args, **unused_kwargs):  # pylint: disable=function-redefined
                        'Use certbot.compat.filesystem.chmod() instead.')
 
 
+# Very similarly to os.open, os.mkdir has the same effect on Windows, to create an unsecured
+# folder. Same mitigation is provided using security.take_ownership and security.chmod.
+# On top of that, we need to handle the fact that os.mkdir is called recursively by os.makedirs.
+# This is done by protecting the original os.mkdir to have the real logic, call it during the
```

Comment:
we no longer call the original `os.mkdir`, making this code much simpler in fact!

Reply:
Comment simplified.